include/fast_float/float_common.h

```diff
@@ -62,13 +62,25 @@ struct value128 {
 /* result might be undefined when input_num is zero */
 fastfloat_really_inline int leading_zeroes(uint64_t input_num) {
 #ifdef FASTFLOAT_VISUAL_STUDIO
+#if defined(_M_X64) || defined(_M_ARM64)
   unsigned long leading_zero = 0;
   // Search the mask data from most significant bit (MSB)
   // to least significant bit (LSB) for a set bit (1).
   if (_BitScanReverse64(&leading_zero, input_num))
     return (int)(63 - leading_zero);
   else
     return 64;
+#else
+    if(!input_num) return 64;
```

Comment:
In the original function there is a slight difference in behavior; the VS version defends the call to `_BitScanReverse64()` with an if, suggesting that 0 is a valid input. However, the gcc version uses `__builtin_clzll()` unprotected, so 0 will pass undetected, but the output will be invalid.

I initially went the VS way. As expected, a test asserting that `leading_zeroes(0) == 64` passes in win64 and win32, and [fails in MinGW](https://github.com/lemire/fast_float/runs/1407301137?check_suite_focus=true#step:7:11).

I'm not sure what to do here. Shall I add an if to MinGW? Shall I remove the ifs and the test?

Reply:
`leading_zeroes(0)` is never called. We do not rely on the fact that `leading_zeroes(0) = 64`. In fact, should 0 be ever be passed to this function, returning 64 would be wrong. There is no way to recover if 0 is passed the function. It cannot and should not happen.

Reply:
We deliberately document it...

```C++
/* result might be undefined when input_num is zero */
```

That is, calling this function with zero is a mistake. 

This should not be checked at runtime (for obvious performance reasons).

Reply:
> This should not be checked at runtime (for obvious performance reasons).

agreed.
```cpp
// Should I then turn this...
if (_BitScanReverse64(&leading_zero, input_num))
  return (int)(63 - leading_zero);
else
  return 64;

// ... into this?
return (int)(63 - _BitScanReverse64(&leading_zero, input_num));
```

Reply:
Looks fine to me.